### PR TITLE
Add before_and_after to moreitertools

### PR DESCRIPTION
**Pull request by danielkza**
### Description

Add before_and_after functin to moreitertools

### CODE REVIEW

**Comment by jreese** [comment]
Please rebase onto latest from main to resolve some of these build issues.

**Comment by danielkza** [comment]
rebased

File: aioitertools/more_itertools.py [hunk 1]
```diff
@@ -44,3 +48,47 @@ async def chunked(iterable: AnyIterable[T], n: int) -> AsyncIterable[List[T]]:
...
+         >>> ''.join([char async for char in remainder])
+         'dEfGhI'
+
+    Note that the first iterator must be fully consumed before the second
+    iterator can generate valid results.
+    """
+
+    it = iter(iterable)
+
+    transition = asyncio.get_running_loop().create_future()
```
Comment (jreese) [inline aioitertools/more_itertools.py:73]: This uses APIs that aren't available on 3.6. `asyncio.get_event_loop()` should be sufficient.

**Review by jreese — CHANGES_REQUESTED** [review]
Overall looks good, just needs some changes to pass CI on 3.6 :)

**Review by jreese — APPROVED** [review]

**Comment by jreese** [comment]
Thank you!

**Pull request merged**